From f9c0578bc95968241fa3dd1f48521239f6a83bca Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Sat, 1 Sep 2018 14:47:01 -0700
Subject: [PATCH 3/3] soong: Add additional_deps attribute for libraries and
 binaries

This sets LOCAL_ADDITIONAL_DEPENDENCIES in the generated makefile.
Takes a list of strings, unverified, so wildcards and variables are
passed, unmodified, directly into the generated makefile.

Change-Id: Ied2bebe4dd80e4292a8090eb5628d83ae52989a1
---
 cc/androidmk.go | 6 ++++++
 cc/binary.go    | 2 ++
 cc/library.go   | 2 ++
 3 files changed, 10 insertions(+)

diff --git a/cc/androidmk.go b/cc/androidmk.go
index 9bcb7835..18b41302 100644
--- a/cc/androidmk.go
+++ b/cc/androidmk.go
@@ -168,6 +168,9 @@ func (library *libraryDecorator) AndroidMk(ctx AndroidMkContext, ret *android.An
 			} else if ctx.useVndk() {
 				fmt.Fprintln(w, "LOCAL_USE_VNDK := true")
 			}
+			if len(library.Properties.Additional_deps) > 0 {
+				fmt.Fprintln(w, "LOCAL_ADDITIONAL_DEPENDENCIES := "+strings.Join(library.Properties.Additional_deps, " "))
+			}
 
 			library.androidMkWriteExportedFlags(w)
 			fmt.Fprintln(w, "include $(BUILD_HEADER_LIBRARY)")
@@ -232,6 +235,9 @@ func (binary *binaryDecorator) AndroidMk(ctx AndroidMkContext, ret *android.Andr
 		if len(binary.Properties.Overrides) > 0 {
 			fmt.Fprintln(w, "LOCAL_OVERRIDES_MODULES := "+strings.Join(binary.Properties.Overrides, " "))
 		}
+		if len(binary.Properties.Additional_deps) > 0 {
+			fmt.Fprintln(w, "LOCAL_ADDITIONAL_DEPENDENCIES := "+strings.Join(binary.Properties.Additional_deps, " "))
+		}
 	})
 }
 
diff --git a/cc/binary.go b/cc/binary.go
index 9e7b70b2..f1516206 100644
--- a/cc/binary.go
+++ b/cc/binary.go
@@ -52,6 +52,8 @@ type BinaryLinkerProperties struct {
 	// binaries would be installed by default (in PRODUCT_PACKAGES) the other binary will be removed
 	// from PRODUCT_PACKAGES.
 	Overrides []string
+
+	Additional_deps []string
 }
 
 func init() {
diff --git a/cc/library.go b/cc/library.go
index 76f8a8cf..9d87b7d9 100644
--- a/cc/library.go
+++ b/cc/library.go
@@ -72,6 +72,8 @@ type LibraryProperties struct {
 	}
 
 	Static_ndk_lib *bool
+
+	Additional_deps []string
 }
 
 type LibraryMutatedProperties struct {
-- 
2.17.1

